clang/test/CodeGenSYCL/simplifycfg.cpp

```diff
@@ -0,0 +1,34 @@
+// RUN: %clangxx -fsycl -fsycl-device-only %s -O3 -S -o - | FileCheck %s
+//
+// Test that SimplifyCFG pass isn't invoked by checking number of function's calls.
+//
+// Usually clang with SimplifyCFG pass optimizes constructs like:
+// if (i % 2 == 0)
+//   func();
+// else
+//   func();
+//
+// into one simple func() invocation.
+// This behaviour might be wrong in cases when func's behaviour depends on
+// a place where it is written.
+// There is a relevant discussion about introducing
+// a reliable tool for such cases: https://reviews.llvm.org/D85603
+
+// CHECK: {{.*}} call spir_func i32 @_Z32__spirv_SubgroupShuffleDownINTELIiET_S0_S0_j
+// CHECK: {{.*}} call spir_func i32 @_Z32__spirv_SubgroupShuffleDownINTELIiET_S0_S0_j
+
+
+#include <sycl.hpp>
+
+int main() {
+  sycl::queue q;
+  int* output = sycl::malloc_shared<int>(1, q);
+  q.parallel_for(sycl::nd_range<1>{1, 1}, [=](sycl::nd_item<1> it){
+    int i = it.get_global_id(0);
+    if (i % 2 == 0) {
+      output[0] = sycl::shift_group_left(it.get_sub_group(), 1, 1);
```

Comment:
To make this work, clang should be able to find `sycl.hpp`. If you build `sycl` target, this header is copied to the build directory and clang is able to pick it up. To expose the problem @AlexeySachkov is talking about you need to build `check-clang` target before building `sycl` target (e.g. `./configure.py` + `./check.py -t check-clang`).

> I suggest that we also have a test similar to [test/CodeGenSYCL/kernel-early-optimization-pipeline.cpp](https://github.com/intel/llvm/blob/sycl/clang/test/CodeGenSYCL/kernel-early-optimization-pipeline.cpp) where we simply check a list of passes added into the pipeline.

Do you suggest having two tests for that issue? I think one test should be enough. Considering that other passes might impact control flow graph as well, just checking that simplifycfg is not added to optimization pipeline doesn't seems to be a good test.

Reply:
> Do you suggest having two tests for that issue? I think one test should be enough. Considering that other passes might impact control flow graph as well, just checking that simplifycfg is not added to optimization pipeline doesn't seems to be a good test.

I was think about creating something closer to unit-test, rather than integration test we have here, but after reading your comment I agree that the current test is indeed better/more reliable probably.